src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptLspFactories.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript;
+
+[ExportLspServiceFactory(typeof(RequestDispatcher), ProtocolConstants.TypeScriptLanguageContract), Shared]
+internal class VSTypeScriptRequestDispatcherFactory : RequestDispatcherFactory
```

Comment:
Nit: In Razor we try to keep it to 1 file in 1 class (unless the second is private or some such), but definitely keep with whatever the standard is in Roslyn.